### ISSUE TITLE
Unexpected response from server crashes ElectrumClient

### ISSUE THREAD

**Issue by dpad85** [ACINQ/lightning-kmp#234]
Connecting to `testnet1.electrum.acinq.co`, my app received an invalid version (should look like `["ElectrumX 1.15.0","1.4"]`).

<summary>Logs
<details>
<p>

```
2021-03-19 18:53:02.376 23561-23561/fr.acinq.phoenix.android E/AndroidRuntime: FATAL EXCEPTION: main
    Process: fr.acinq.phoenix.android, PID: 23561
    java.lang.IllegalArgumentException: Element class kotlinx.serialization.json.JsonObject (Kotlin reflection is not available) is not a JsonArray
        at kotlinx.serialization.json.JsonElementKt.error(JsonElement.kt:237)
        at kotlinx.serialization.json.JsonElementKt.getJsonArray(JsonElement.kt:167)
        at fr.acinq.eclair.blockchain.electrum.ElectrumDataTypesKt.parseJsonResponse(ElectrumDataTypes.kt:219)
        at fr.acinq.eclair.blockchain.electrum.WaitingForVersion.process(ElectrumClient.kt:64)
        at fr.acinq.eclair.blockchain.electrum.ElectrumClient.run(ElectrumClient.kt:205)
        at fr.acinq.eclair.blockchain.electrum.ElectrumClient$run$1.invokeSuspend(Unknown Source:11)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7660)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```
</p></details></summary>

Unfortunately I don't have the actual json sent by the server.

When that happens, we should print the faulty response and disconnect from the server. Not sure if we should handle that here or in phoenix.

**Comment by sstone** [comment]
I cannot reproduce on linux, maybe an issue with the low-level connection library ?

**Comment by dpad85** [comment]
It's not systematic, I cannot reproduce it either. I guess that sometimes electrum does not send the expected data (maybe we expect to receive the version first?).

**Comment by t-bast** [comment]
Fixed by #235
